### ISSUE TITLE
Multiple selected option in creatable caused by auto trim spaces

### ISSUE THREAD

**Issue by zyxneo** [JedWatson/react-select#4491]
Hi!

To reproduce the issue visit https://react-select.com/creatable.

1. In **Advanced Example** type: "a b" ( A SPACE B)
2. Clicking `Create "a b"` button
3. In the same field type: "a   b" ( A SPACE SPACE B)
4. Clicking `Create "a b"` button (notice that feedback is trimmed already)
5. Open the dropdown

Notice, that two option is selected, and both labeled `a b`.
Any extra space added in between or after the same sequence of strings causes the same behavior and adds further selected options with the same label.
(The editor removes the spaces from this comment.)

**Comment by Rall3n** [comment]
Hello @zyxneo,
the behavior you see as an error is intentional for the example you referenced (the `CreatableAdvanced` example).

Take a look at the [example code](https://github.com/JedWatson/react-select/blob/master/docs/examples/CreatableAdvanced.js#L10-L13).

The function `createOption` is a custom function which is called every time an option is created. It takes the entered text and turns it into the new option.

When it is called it straight up uses the entered text as the `label`. For the `value` it lowercases the text and removes everything that is not part of a word (RegEx `/\W/g`), including spaces.

Based on your example the text `a b` would be turned into the value `ab` . The same goes for the second variant with two spaces. This explains why both are selected even though you created a different option.

---

Regarding the visible option text (the `label`): That is part of the [offical HTML specification](https://www.w3.org/TR/REC-html40/struct/text.html#h-9.1):
> [...] Note that a sequence of white spaces between words in the source document may result in an entirely different rendered inter-word spacing (except in the case of the PRE element). In particular, user agents should collapse input white space sequences when producing output inter-word space.

In other words: Multiple spaces in a row will be rendered as if there is only one space. But this can be countered either using the `<pre>` element or the CSS property `white-space: pre;`.

---

Hope I could give some clarification on why this happens.

**Comment by zyxneo** [comment]
Dear @Rall3n !

Many thanks for the detailed explanation! It makes sense, and it helps. So practically the mentioned issue should not be a use-case in a real-life scenario.

Many thanks again for your patience and time.

BR!

**Comment by Dem0n3D** [comment]
>The function createOption is a custom function which is called every time an option is created. It takes the entered text and >turns it into the new option.
>
>When it is called it straight up uses the entered text as the label. For the value it lowercases the text and removes everything >that is not part of a word (RegEx /\W/g), including spaces.

This example is awful: it removes all non-latin characters.
